docs(bodiless-documentation): Add Site Build Guide (#131)
johnsonandjohnson/Bodiless-JS: packages/bodiless-documentation/doc/Development/Guides/BuildingSites/ComponentsStarterKit.md

> # Components & Tempaltes that come in the Bodiless starter

Reviewer: Tempaltes -> Templates

Author: addressed

  withMetaHtml('en'),
)(Helmet);
```
`withMeta`, `withMetaTitle` are HOC's that will obtain the value of data out json object associated at page level or if specified at site level.

e.g.
- `withMeta('pagetype', 'page-type')` will obtain its value at page level from `meta$page-type.json` which contains `{	"content": "Home" }`
- `withMeta('bl-brand', 'brand', 'site')` will obtain its value from `src/data/site.meta$brand.json` which contains `{ "content": "BodilessJS" }`

### Site Title & Logo

* This can be configured in `gatsby-config.js` by setting SiteMetaData.

```
  siteMetadata: {
    title: 'BodilessJS Starter',
    logo: '/images/bodiless_logo.png',
  },
```
Note: the mobile logo is defined within `/src/components/Menus/token.tsx` and
doesn't use siteMetaData.

### Site Favicon

* Can be updated in `src/components/Layout/index.jsx` by defining location of
  favicon.

### Addition of other components in the Header & Footer

Often there are other elements in the header/footer and they can be added by
manipulating these files. As you add more components, remember the header and
footer are components that all pages will use, so the data for them should be

> stored at site level. This can be done via specifying `nodeCollect="site"`.

Reviewer: nodeCollect -> nodeCollection

Author: addressed

# Components & Templates that come in the Bodiless starter

Let us discuss a few of these components/templates that are contained within the Bodiless starter:

## Layout

These components consist of the global components

* Header `src/components/Layout/header.tsx`
* Footer `src/components/Layout/footer.tsx`
* Layout `src/components/Layout/index.jsx`
  * This file bring together header & footer into the page layout & composes the
    metadata of the page.

These components work as-is out of the Bodiless starter but you may have to adjust
their tokens (styling) to meet the designs of the site.

> ### MetaData Component

Author: ok

> ### MetaData Component

Reviewer 2: how do you want to handle this?  should we merge this and then make sure he updates the docs in scope of his ticket?

> ### MetaData Component

Author: I am going to redo the metadata here based on what @dewen  has done so far and I will discuss with him updating it if anything changes.

> Within `src/components/Layout/index.jsx` you will find Helmet component that

Author: updated

will manage all of your metadata in the document head.

The starter kit provides the following example:
```
const ExampleHelmet = flowRight(
  asBodilessHelmet('meta'),
  withMeta('pagetype', 'page-type'),
  withMeta('description', 'description'),
  withMeta('bl-brand', 'brand', 'site'),

>   withMeta('bl-country', 'country', 'site'),

Author: To be discussed...

>   withMeta('bl-country', 'country', 'site'),

Reviewer 2: ok you can resolve this once you've added it

>   withMeta('bl-country', 'country', 'site'),

Author: It's added to the Starter Kit story https://github.com/johnsonandjohnson/Bodiless-JS/issues/459

  withMetaTitle('page-title'),
  withMetaHtml('en'),
)(Helmet);
```
`withMeta`, `withMetaTitle` are HOC's that will obtain the value of data out
json object associated at page level or if specified at site level.

e.g.
- `withMeta('pagetype', 'page-type')` will obtain its value at page level from
  `meta$page-type.json` which contains `{ "content": "Home" }`
- `withMeta('bl-brand', 'brand', 'site')` will obtain its value from
  `src/data/site.meta$brand.json` which contains `{ "content": "BodilessJS" }`

### Site Title & Logo

* This can be configured in `gatsby-config.js` by setting SiteMetaData.

```
  siteMetadata: {
    title: 'BodilessJS Starter',

>     logo: '/images/bodiless_logo.png',

Author: To be discussed...

>     logo: '/images/bodiless_logo.png',

Reviewer 2: ok - resolve this convo once you update the story.

>     logo: '/images/bodiless_logo.png',

Author: It's added to the Starter Kit story https://github.com/johnsonandjohnson/Bodiless-JS/issues/459

  },
```
Note: the mobile logo is defined within `/src/components/Menus/token.tsx` and

> doesn't use siteMetaData.

Author: To be discussed...

> doesn't use siteMetaData.

Reviewer 2: ok - resolve this convo once you update the story.

> doesn't use siteMetaData.

Author: It's added to the Starter Kit story https://github.com/johnsonandjohnson/Bodiless-JS/issues/459

### Site Favicon

* Can be updated in `src/components/Layout/index.jsx` by defining location of
  favicon.

### Addition of other components in the Header & Footer

Often there are other elements in the header/footer and they can be added by
manipulating these files. As you add more components, remember the header and

> footer are components that all pages will use, so the data for them should be

Author: @wodenx  I would to nodecollection if was documented. :)

stored at site level. This can be done via specifying `nodeCollection="site"`.

e.g.
```

>  <Menu nodeKey="MainMenu" nodeCollection="site" />

Author: To be discussed...

>  <Menu nodeKey="MainMenu" nodeCollection="site" />

Reviewer 2: ok - resolve this convo once you update the story.

>  <Menu nodeKey="MainMenu" nodeCollection="site" />

Author: It's added to the Starter Kit story https://github.com/johnsonandjohnson/Bodiless-JS/issues/459

```

## Menu & Burger Menu

The Bodiless starter comes with menu that is placed in the header. It is menu
that the data is shared site-wide so that all pages use the same menu data. In
addition the burger menu (mobile) menu generates its menu off this desktop
menu's data. For more details, please see
[Menu](/Components/Organisms/MainMenu) &
[Burger Menu](/Components/Organisms/BurgerMenu)

## Editors

The Bodiless starter comes with pre-defined editors that can be changed and/or
extended to meet the site requirements. You will find three rich text editors
defined that are Simple, Basic & FullFeatured.

* Simple : Superscript Only
* Basic : Superscript, Bold, Italic, Underline, Link, Align Text
* Full Featured : adds in more functionality to Basic such as Strikethru, headers, quotes.

> 

Reviewer 2: Insert here: "Editors can be extended or customized by using the [Bodiless Design API](link) to apply token HOC's which add, remove, replace or style the components which render different text formatting.

Author: Addressed

We recommend
[refining the editor components](/Components/RichText?id=richtext-component)
needed to meet your site's requirements.

## Images

> The images contain simple editable images that are either render linkable or

Reviewer 2: The starterkit prov
```suggestion
The starterkit provides simple editable images that render either linkable or
```

Author: addressed

nonlinkable images and a few variations of image placeholders.

For more information please read [Images](/Components/Image).

## Touts

Most sites use some form of Touts. (A block that consists of Image, Title, Body
and Call to Action) The Bodiless starter comes with this basic form defined in
`src/components/Tout`. These can be used as is and/or extended to meet
the site design requirements. This component has its own set of tokens as well
for its different variations.

## Flow Container

> The Flow Container component is a layout component that allows you to select from

Reviewer 2: ```suggestion
The Flow Container component is a layout component that allows a content editor to select from
```

Reviewer 2: Generally - I think go through and be consistent about the pronounds and personas.  I think "you" should always be the site builder.

Author: Resolved

a set of components, place them on the page, and resize them.

> The Editors, Images & Touts are available as a starting point of components that
> can be added to your site via the Flow Container. Additional components as they
> are built or extended can be added to allow other components to be added via
> FlowContainer. In addition, you can create multiple variations of FlowContainers
> to control what components are used in different areas of sites. e.g. A
> FlowContainer for a sidebar could have reduced set of components to select from.

Reviewer 2: ```suggestion
You define set of component variations available to be placed in any particular FlowContainer.
The starter kit provides a starting point which contains the example components described
above (Editors, Images and Touts).  As with editors, you can use the design api to add, remove,
replace or style these variations. In addition, you can create different FlowContainers
to control what components may be placed in different areas of sites. e.g. A
FlowContainer for a sidebar could have reduced set of components to select from.
```

Author: Addressed

For more information, please read [FlowContainer](/Components/FlowContainer).

## Sitemap.xml & Robots.txt file

Following best practices, the site will automatically render the sitemap.xml &

> robots.txt file.

Reviewer 2: Do we give any ability to customize?

Author: @wodenx  We have some documentation but its over in CanvasX and I can't link there.. So for now I left it as this.

## Templates

> A default template can be found in `src/templates/_default.jsx` that uses the FlowContainer.

Reviewer 2: ```suggestion
A default page template using a FlowContainer can be found in `src/templates/_default.jsx` 
```

Author: addressed

* Header `src/components/Layout/header.tsx`
* Footer `src/components/Layout/footer.tsx`
* Meta `src/components/Layout/meta.tsx`
* Logo `src/components/Layout/logo.tsx`
* Layout `src/components/Layout/index.tsx`
  * This file bring together header & footer into the page layout & composes the
    metadata of the page.

These components work as-is out of the Bodiless starter but site builder may
have to adjust their tokens (styling) to meet the designs of the site.

### MetaData Component

Bodiless provides a set of HOC's which work with react-helmet to place editable
meta-tags in the document HEAD.  A Site Builder can find an example in
`src/components/Layout/meta.tsx`.

For full code, please
[review code](https://github.com/johnsonandjohnson/Bodiless-JS/tree/master/examples/starter/src/components/Layout)

> The components within this file are providing two abilities:

Reviewer 2: ```suggestion
The HOC's used in this file are responsible for both rendering and editing meta-tags:
```
```

#### Adding SEO form to Editor interface
The `withMetaForm` provides ab ability to insert a SEO form
within the editor inferface for the site editor to manipulate meta date per
page. 

Besides adding the form, the site builder will also define the fields that the

> site builder can see to edit in the page.

Reviewer 2: ```suggestion
content editor can see to edit in the page.
```

#### Adding metatag to the document HEAD

> In addition, the calls to `withMeta*` are HOC's that will obtain the value of

Reviewer 2: ```suggestion
In addition to defining the form fields, the calls to `withMeta*` also obtain the value of
```

> In addition, the calls to `withMeta*` are HOC's that will obtain the value of
> data out json object (where its written by the editor.) You can also review and
> see some meta data are associated at page level or at a site (global) level.

Reviewer 2: ```suggestion
In addition fo defining the form fields, the calls to `withMeta*` also render
the meta-tags to the document head, using data from json objects which were
written by the editor. You can also see defined here some site-level meta-tags
which are not exposed to the editor for modification.
```
